activerecord/lib/active_record/connection_adapters/mysql2_adapter.rb

```diff
@@ -141,6 +141,7 @@ def connect
 
         def reconnect
           @raw_connection&.close
+          @raw_connection = nil
```

Comment:
I'm questioning whether this is actually necessary, since the next line calls `connect`, which immediately reassigns `@raw_connection` to something else

Reply:
oh, but if `connect` fails, we will be stuck with the old connection. So better set it to `nil`. I understand now